gtfs-realtime/spec/en/reference.md

```diff
@@ -215,7 +220,7 @@ The relation between this StopTime and the static schedule.
 |-------------|---------------|
 | **SCHEDULED** | The vehicle is proceeding in accordance with its static schedule of stops, although not necessarily according to the times of the schedule. This is the **default** behavior. At least one of arrival and departure must be provided. Frequency-based trips (GTFS frequencies.txt with exact_times = 0) should not have a SCHEDULED value and should use UNSCHEDULED instead. |
 | **SKIPPED** | The stop is skipped, i.e., the vehicle will not stop at this stop. Arrival and departure are optional. When set `SKIPPED` is not propagated to subsequent stops in the same trip (i.e., the vehicle will stop at subsequent stops in the trip unless those stops also have a `stop_time_update` with `schedule_relationship: SKIPPED`). Delay from a previous stop in the trip *does* propagate over the `SKIPPED` stop. In other words, if a `stop_time_update` with an `arrival` or `departure` prediction is not set for a stop after the `SKIPPED` stop, the prediction upstream of the `SKIPPED` stop will be propagated to the stop after the `SKIPPED` stop and subsequent stops in the trip until a `stop_time_update` for a subsequent stop is provided.  |
-| **NO_DATA** | No data is given for this stop. It indicates that there is no realtime timing information available. When set NO_DATA is propagated through subsequent stops so this is the recommended way of specifying from which stop you do not have realtime timing information. When NO_DATA is set neither arrival nor departure should be supplied. |
+| **NO_DATA** | No data is given for this stop. It indicates that there is no realtime timing information available. When set NO_DATA is propagated through subsequent stops so this is the recommended way of specifying from which stop you do not have realtime timing information. When NO_DATA is set neither arrival nor departure should be supplied. NO_DATA must not be used for added or replacement trips, as the StopTimeUpdate defines the stop list of the trip. |
```

Comment:
Not allowing NO_DATA to be used for ADDED or REPLACEMENT trips means that producers can output ADDED or REPLACEMENT trips only when they have real-time data available for the trip (SCHEDULED = "The vehicle is proceeding in accordance with its static schedule of stops..."). With the addition of `scheduled_time`, there is an opportunity to allow consumers to define these ADDED or REPLACEMENT trips before real-time data is available. This would allow producers to provide valuable data to consumers earlier so customers get the info sooner. What do others think?

Reply:
I think your idea is good, and I have to change the definition of `NO_DATA` to "no real-time data is given for this stop".